### ISSUE TITLE
Function of `first_frame_tokens.detach()` before decoder call

### ISSUE THREAD

**Issue by sroutray** [LatentActionPretraining/LAPA#11]
https://github.com/LatentActionPretraining/LAPA/blob/33a2db32957dd8ccbb709de82ad96fb04d7fa57c/laq/laq_model/latent_action_quantization.py#L268

It seems that you detach the `first_frame_tokens` before using at as conditioning feature in cross attention in decoder. Could you confirm if this is intentional? It seems to break the gradient flow back to token embedding layer while training.

![image](https://github.com/user-attachments/assets/d2d024e8-ddfd-45f7-9ed5-a4cf33be137e)

**Comment by seonghyeonye** [comment]
Yes, we apply a stop gradient to the first frame patch embedding to avoid representation collapse. We will modify the Figure to avoid misunderstanding.
